### DISCUSSION TITLE
用webdav无法下载，但是在网页端可以下载

### DISCUSSION THREAD

**Discussion by zjw345** [AlistGo/alist#532] [Q&A] [answered]
用webdav无法下载可以上传，但是在前端网页是可以下载的，是什么原因呢？
是在https://alist-doc.nn.ci/docs/driver/aliyundrive/ 扫码获取的refresh_token。
在网页可以下载，应该就不是网盘的原因吧，是因为Alist的webdav服务的问题吗？
![微信图片_20220213164243](https://user-images.githubusercontent.com/35001234/153745685-d776f77e-45e4-4e11-ac38-88d63223d34e.png)
![微信图片_20220213164231](https://user-images.githubusercontent.com/35001234/153745697-0c3d4eca-cfff-47fe-a739-0b59e6dd120b.png)

**Comment by xhofe** [comment, marked as answer]
我没有nas，不知道cloud sync是否支持302跟随， 目前webdav下载使用的方式是302跳转，所以需要客户端支持 302跟随。
下个版本会支持webdav下载直传： https://github.com/Xhofe/alist/commit/4371c470b3224d8650e2b379d122b9bca7b7bd32

**Reply by zjw345** [reply]
对webdav有几种方式？
经测试：
挂载**天翼云盘**使用webdav连接，**cloud sync能**下载，但是**RaiDrive不能**下载，上次均无问题。
挂载**阿里云盘**使用webdav连接，**cloud sync不能**下载，但是**RaiDrive能**下载，上次均无问题。
是什么原因呢，有点抓狂，需要怎么才能完美解决兼容？

**Reply by zjw345** [reply]
打错字，上传均无问题，

**Reply by xhofe** [reply]
如果天翼云盘可以下载的话，那应该是支持302跟随的。
这个有更详细的错误信息吗？比如下载的时候返回的是什么内容？
或者你在阿里的账号那里打开webdav代理试一下可以下载吗？

**Comment by solonJiang** [comment]
v2.5.2版本打开webdav direct 就可以下载了。

**Comment by kooyeed** [comment]
我打日志发现，客户端如果直接302的话，阿里云盘可能禁掉。我在alist开代理就没有问题了。

**Reply by kooyeed** [reply]
我在客户端挂载(rclone mount）的时候增加了一个参数就可以302了,希望对大家有帮助。

--header "Referer:https://www.aliyundrive.com/"

**Reply by MichaelChao02** [reply]
> 我在客户端挂载(rclone mount）的时候增加了一个参数就可以302了,希望对大家有帮助。
> 
> --header "Referer:https://www.aliyundrive.com/"

遇到相同的问题，这个方法完美解决了问题！

**Reply by yhx123** [reply]
> 我在客户端挂载(rclone mount）的时候增加了一个参数就可以302了,希望对大家有帮助。
> 
> --header "Referer:https://www.aliyundrive.com/"

你好，问一下老铁怎么加这个请求头呀
